flutter/lib/src/integrations/screenshot_integration.dart

```diff
@@ -0,0 +1,29 @@
+import 'dart:async';
+
+import 'package:flutter/scheduler.dart';
+import 'package:sentry/sentry.dart';
+import '../screenshot/screenshot_attachment_processor.dart';
+import '../sentry_flutter_options.dart';
+
+class ScreenshotIntegration implements Integration<SentryFlutterOptions> {
+  SentryFlutterOptions? _options;
+
+  @override
+  FutureOr<void> call(Hub hub, SentryFlutterOptions options) {
+    // ignore: invalid_use_of_internal_member
+    options.clientAttachmentProcessor = ScreenshotAttachmentProcessor(() {
+      try {
+        /// Flutter >= 2.12 throws if SchedulerBinding.instance isn't initialized.
+        return SchedulerBinding.instance;
```

Comment:
We can use the `SchedulerBindingProvider` similar to `NativeAppStartIntegration`

Reply:
This is pretty much the same approach, or are you refering to something else? https://github.com/getsentry/sentry-dart/blob/322aa66508dd3f9d0f68edacb140bc1f9f53e3f5/flutter/lib/src/sentry_flutter.dart#L157

Reply:
oh nvm, its a typedef and you are defining the callback.